### ISSUE TITLE
Ghosts of Saltmarsh

### ISSUE THREAD

**Issue by LordMorin** [Sklore/HL_DD_5e_Colab#901]
Opening this to add a Source for the Ghosts of Saltmarsh. At the very least I will be adding the new backgrounds. I don't know if the new Vehicles are even possible in Hero Lab at this time.

**Comment by LordMorin** [comment]
New Source for Ghosts of Saltmarsh should now be added.

**Comment by Fenris447** [comment]
I’m working on getting the basic monster information in using the same process daplunk did for MTOF. I’m also working on a bunch of standardized racial/monster abilities using the {abtext} replacement method that should make adding abilities streamlined for all our outstanding monsters.

**Comment by Fenris447** [comment]
We could add the ships, using abilities to show all their extra statistics. I haven’t read into the whole system of stats but I’d like to figure out a way to at least track their HP within the tactical console.

**Comment by LordMorin** [comment]
My room mate needs the backgrounds, so I figured I'd make them available here when I am done.

**Comment by LordMorin** [comment]
Added the 4 new backgrounds:
Fisher, Marine, Shipwright, & Smuggler
- For these Backgrounds only, I've added in the Saltmarsh Ties. I am not sure if its worth the codign effort of addign the Saltmarsh Ties for the older backgrounds . . .

**Comment by LordMorin** [comment]
Actually Merged the Source and Background files . . . <duh>

**Comment by Fenris447** [comment]
We could make copies of the older backgrounds and add in the Saltmarsh ties to them. Then we set the source for those copies to GoS so they don’t show up again. I can take care of it.

**Comment by LordMorin** [comment]
Feel free. :)

**Comment by Fenris447** [comment]
Looking at them, it would be pretty easy to at least add everything that’s in a table, without even needing to create extra copies of the backgrounds.

**Comment by LordMorin** [comment]
I was thinking of an Adjustment that reads your background and adds the Saltmarsh Tie info to the Background's description text . . .

**Comment by LordMorin** [comment]
Added the magic Items.

**Comment by Fenris447** [comment]
<img width="473" alt="GoS Source Error" src="https://user-images.githubusercontent.com/32333990/58842719-2265db00-8635-11e9-813c-5061109fffe3.PNG">

I downloaded the source file and am getting this. I looked through the code and can't find anything wrong. :(

**Comment by LordMorin** [comment]
Strange. I looked at the code I had uploaded to GIT and it all looks fine.
I am usign the same code atm without issues.

Is it possible my file is conflicting with private code on your system?

**Comment by Fenris447** [comment]
I deleted my comment because I realized I did something dumb. The problem was me, not the file. Looks good!

**Comment by LordMorin** [comment]
np, I get my share of such actions all the time :)

**Comment by Fenris447** [comment]
I’ve copyright-safe rewritten the parts of the backgrounds (Outlander, Noble, etc) that are just additional options when rolling on the tables. They’re set to use GoS as a source so they’ll only show up when it’s selected. The original options will still be there too, though.

If you can get a framework done for how the adjustment for all the new Ties to the backgrounds would work, I can do the grunt work on rewriting and adding them.

Basically if you give me the adjustment code that says “look for background X, replace Y with Z,” I can do the dirty work.

X = existing background
Y = its original description
Z = the new Saltmarsh tie

**Comment by LordMorin** [comment]
Had I uploaded my files and you just made the text changes? Or did you create the backgrounds yourself and I forget to upload mine?

**Comment by Fenris447** [comment]
You uploaded your files. I pulled them and have been making additions to the one you made for backgrounds. I haven’t uploaded those changes yet.

**Comment by LordMorin** [comment]
kk, thanks. I thought I hadn't uploaded what I had. :)

For the past 4 days I've been trying to get my mind around to what I need to do, but I have been unable to focus . . . <sigh>
I know what I need to get done, but just haven't been able to get my mind around it. :(

**Comment by Fenris447** [comment]
I have almost all of the abilities for the GoS monsters programmed. As I did with MTOF, my process is to build each monster as I build the .stock file one monster at a time, associating their abilities with them as I go. I’m simultaneously trying to add in every missing monster from Tyranny of Dragons, so it’ll all probably be finished around the same time. And I promise it will be before 2022. :D

**Comment by Fenris447** [comment]
Every ability is programmed, and about 2/3 of the monsters and the stock file have been completed. Tyranny of Dragons is now complete so this is getting 80% of my attention (the other 20% is Eberron 😄).

**Comment by Fenris447** [comment]
All the monsters are done and uploaded here to my current branch. I think there are a few items and some background work left. Then all that's not in will be Ships.

**Comment by Fenris447** [comment]
I'm working on vehicles right now. I have the Galley almost completed, using the in-play tab to track the HP of individual components. It was either that, or build each component as an individual minion that then fed back to the Hero ship. And that was just too much work for too little gain. I built the components to be mostly interchangable, so the rest of the ships will go fast. And the BGDIA war machines are even simpler, so we may have these ready within the next release or two.

**Comment by Fenris447** [comment]
Vehicles are donezo and will be in release 3.0

**Comment by Fenris447** [comment]
Turns out I already finished the backgrounds and items. So we're done with GoS. Booyah.
